testing/trino-tests/src/test/java/io/trino/tests/AbstractTestEngineOnlyQueries.java

```diff
@@ -6124,6 +6124,46 @@ public void testJoinWithNonOrderableType()
                 .matches("SELECT CAST(map(ARRAY['name'], ARRAY['trino']) AS map(varchar, varchar))");
     }
 
+    /**
+     * See https://github.com/trinodb/trino/issues/9171
+     * <p>
+     * Only occurs in distributed planning mode
+     */
+    @Test
+    public void testPartialLimitWithPresortedConstantInputs()
+    {
+        assertThat(query("SELECT a\n" +
+                "FROM (\n" +
+                "    SELECT 0, 1\n" +
+                "    FROM (\n" +
+                "        SELECT 1\n" +
+                "        FROM (VALUES (1, 1, 1)) t(k, g, h)\n" +
+                "            CROSS JOIN (VALUES 1)\n" +
+                "        GROUP BY k\n" +
+                "    )\n" +
+                "    UNION ALL\n" +
+                "    SELECT 0, 1\n" +
+                ") u(a, b)\n" +
+                "ORDER BY b\n" +
+                "LIMIT 10"))
+                .matches("VALUES (0), (0)");
+
+        assertThat(query("SELECT *\n" +
+                "FROM (\n" +
+                "    VALUES (0, 1)\n" +
```

Comment:
worth adding a test based on tables; in the future we may optimize more and more around VALUES (like inline GROUP BY over VALUES), making the test void

Reply:
I couldn't reproduce it with a table. Even minor variations in that query makes it stop failing.

Reply:
I had repro'ed it with a table
```
CREATE TABLE partitioned_lineitem WITH (partitioned_by=array['suppkey', 'linenumber']) AS SELECT orderkey, partkey, (suppkey * 0) suppkey, (linenumber * 0) linenumber FROM tpch.tiny.lineitem;

select partkey, suppkey, linenumber from partitioned_lineitem where suppkey=0 and linenumber=0 and partkey=1 order by partkey limit 10;
```

Reply:
I was able to reproduce it with this query against the TPC-H connector:

```sql
SELECT orderkey, custkey 
FROM orders 
WHERE orderkey = 1 AND custkey = 370 
ORDER BY orderkey 
LIMIT 1
```

Still, that's somewhat brittle, as it only fails if the engine decides to order the local properties for the table as `C(custkey), C(orderkey)` and not the other way around, which is equivalently valid and would make the query pass.